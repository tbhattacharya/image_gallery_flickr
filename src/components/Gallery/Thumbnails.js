import React from 'react';
import { LazyLoadImage } from 'react-lazy-load-image-component';
import { Thumbs } from '../../utils/Constants';
import 'react-lazy-load-image-component/src/effects/blur.css';
import Grid from '@material-ui/core/Grid';
import { withStyles } from '@material-ui/core/styles';
import { styles } from './GalleryStyles';

class Thumbnails extends React.Component {

    handleClick = ()=>{
        this.props.handleClick(this.props.image);
    }
    render() {
        const { image } = this.props;
        //Create image url for Lazy loader and return grid item element
        const src = `https://farm${image.farm}.staticflickr.com/${image.server}/${image.id}_${image.secret }.jpg`;
        return (
            <Grid item xs={3} classes={{ container: 'true' }} className={this.props.classes.thumbnail}
                onClick={this.handleClick}>
                <div className={this.props.classes.overlayItem} >
                    {image.title}
                </div>
                <div className={this.props.classes.photo}>
                    <LazyLoadImage
                        height={Thumbs.height}
                        src={src}
                        effect="blur"
                        width={Thumbs.width} />
                </div>
            </Grid>
        );
    }
}
export default withStyles(styles)(Thumbnails);