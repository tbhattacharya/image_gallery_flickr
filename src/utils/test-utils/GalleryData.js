export const photo = [
    {
        "id": "24173905193",
        "owner": "75857967@N08",
        "secret": "89196bf532",
        "server": "1695",
        "farm": 2,
        "title": "Cuando amanece la pasi√≥n",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 1,
        "has_comment": 0
    },
    {
        "id": "23971366330",
        "owner": "127665714@N08",
        "secret": "197b0b3cc5",
        "server": "1675",
        "farm": 2,
        "title": "The Lonely Tree",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "24738296325",
        "owner": "78621811@N06",
        "secret": "9891e3ea56",
        "server": "1618",
        "farm": 2,
        "title": "Zig Zag",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "25432433465",
        "owner": "47919595@N02",
        "secret": "472cd8795a",
        "server": "1535",
        "farm": 2,
        "title": "Lake Mungo / Mungo National Park",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "28028506096",
        "owner": "42957889@N05",
        "secret": "f6a6eb1a10",
        "server": "7433",
        "farm": 8,
        "title": "A Wonderful Day",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "27853439812",
        "owner": "113231763@N07",
        "secret": "1b886779f3",
        "server": "7299",
        "farm": 8,
        "title": "Tropic Thunder II",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "24679948091",
        "owner": "62118446@N03",
        "secret": "dd3e3f117f",
        "server": "1671",
        "farm": 2,
        "title": "Braving the cold",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "26305213826",
        "owner": "138050950@N05",
        "secret": "5fa1957826",
        "server": "1645",
        "farm": 2,
        "title": "Sound of silence [explored]",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "27190173775",
        "owner": "130724719@N05",
        "secret": "e9b9533ff4",
        "server": "7156",
        "farm": 8,
        "title": "French Senate from the other side !",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "26694616403",
        "owner": "137893408@N06",
        "secret": "6904ee640f",
        "server": "7628",
        "farm": 8,
        "title": "On the road",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "28302776541",
        "owner": "41993130@N04",
        "secret": "6f9c2a6d28",
        "server": "7654",
        "farm": 8,
        "title": "Sunset - Grand Teton National Park",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "25933072772",
        "owner": "116424075@N06",
        "secret": "bbb1a3e60c",
        "server": "1623",
        "farm": 2,
        "title": "spotlight",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "28099637150",
        "owner": "114727552@N04",
        "secret": "f6872239ba",
        "server": "8397",
        "farm": 9,
        "title": "Changing of the guard",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "28040789350",
        "owner": "55910595@N05",
        "secret": "bed6293793",
        "server": "8853",
        "farm": 9,
        "title": "Tempest || Dubai",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "24465232589",
        "owner": "30125007@N03",
        "secret": "695c05f11e",
        "server": "1636",
        "farm": 2,
        "title": "Bringing back Autumn",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "24272982102",
        "owner": "70992083@N07",
        "secret": "612bf3f919",
        "server": "1571",
        "farm": 2,
        "title": "Beyond Perception",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "24881801442",
        "owner": "91805169@N04",
        "secret": "dd850d575a",
        "server": "1651",
        "farm": 2,
        "title": "Misty Forest",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "23623259713",
        "owner": "130984525@N03",
        "secret": "27480b6928",
        "server": "1713",
        "farm": 2,
        "title": "Infinity",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "27591124962",
        "owner": "35591378@N03",
        "secret": "946ba99eb6",
        "server": "7124",
        "farm": 8,
        "title": "P061516PS-0016",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "24354970986",
        "owner": "123495891@N02",
        "secret": "042477cac3",
        "server": "1649",
        "farm": 2,
        "title": "Superman & Lex Luthor",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "25659172370",
        "owner": "43738607@N07",
        "secret": "ee3b364102",
        "server": "1705",
        "farm": 2,
        "title": "i.get.a.kick.out.of.you",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "24246227076",
        "owner": "136485307@N06",
        "secret": "5f8bc64d74",
        "server": "1523",
        "farm": 2,
        "title": "Apollo 8 Liftoff View",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "24519079014",
        "owner": "57803084@N07",
        "secret": "7ebb3d9af7",
        "server": "1532",
        "farm": 2,
        "title": "",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "25532790890",
        "owner": "29021614@N05",
        "secret": "c261f84f72",
        "server": "1652",
        "farm": 2,
        "title": "Rainbow Ridge",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "25768216394",
        "owner": "15599795@N06",
        "secret": "5a2e6a03a7",
        "server": "1491",
        "farm": 2,
        "title": "Freedom !",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "25017672329",
        "owner": "45992839@N04",
        "secret": "e5b9967612",
        "server": "1659",
        "farm": 2,
        "title": "Red Barn | Norway",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "26092060062",
        "owner": "68778436@N04",
        "secret": "f2819c0c4e",
        "server": "1587",
        "farm": 2,
        "title": "Flickr Top 50, Top US & Cover Photo 2016 (Fetch - Sunset)",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "27131116894",
        "owner": "112386875@N05",
        "secret": "465d9d7c8a",
        "server": "7431",
        "farm": 8,
        "title": "Mountain Glory (Explore)",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "28257256940",
        "owner": "96106108@N07",
        "secret": "318ec718f9",
        "server": "8156",
        "farm": 9,
        "title": "frames",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "27783727896",
        "owner": "52725579@N05",
        "secret": "6e10e7837d",
        "server": "7672",
        "farm": 8,
        "title": "Wallaby Sunset (Explored Jun 21, 2016)",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "24890975982",
        "owner": "66926252@N07",
        "secret": "282deddba9",
        "server": "1599",
        "farm": 2,
        "title": "Bow Valley Sunrise",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "24393016550",
        "owner": "34430197@N04",
        "secret": "127351d623",
        "server": "1572",
        "farm": 2,
        "title": "Sunset From The Cave",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "29214702851",
        "owner": "22461382@N00",
        "secret": "7feb8d9b34",
        "server": "8107",
        "farm": 9,
        "title": "f o g ¬∑ f i n g e r s | marin county, california",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "26258214422",
        "owner": "134276106@N02",
        "secret": "e75fed0ca9",
        "server": "1477",
        "farm": 2,
        "title": "Every Forgotten Dream",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "23822677670",
        "owner": "83802319@N00",
        "secret": "ea2fc01f71",
        "server": "1616",
        "farm": 2,
        "title": "2016 January Fuji",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "30447951876",
        "owner": "78422747@N04",
        "secret": "dec0cae941",
        "server": "5644",
        "farm": 6,
        "title": "Autumn Arch",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "29133157664",
        "owner": "28164079@N05",
        "secret": "856aef9bc3",
        "server": "8385",
        "farm": 9,
        "title": "Norwegian romantic nationalism",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "26723045205",
        "owner": "18205706@N06",
        "secret": "6fc0847622",
        "server": "1519",
        "farm": 2,
        "title": "Ballistic",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "23761130783",
        "owner": "33616145@N00",
        "secret": "843132bdc5",
        "server": "1611",
        "farm": 2,
        "title": "..and then the sky went pink",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "29194948535",
        "owner": "91140701@N03",
        "secret": "48ac690a0c",
        "server": "8544",
        "farm": 9,
        "title": "Kenya sunset (On Explore 08/24/2016)",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "26230731605",
        "owner": "57866871@N03",
        "secret": "4c10757ae2",
        "server": "1472",
        "farm": 2,
        "title": "Brixham twilight",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "27188420796",
        "owner": "28319834@N05",
        "secret": "1154494306",
        "server": "7187",
        "farm": 8,
        "title": "Morning reflections",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "26957168394",
        "owner": "40252249@N00",
        "secret": "30e3ac4250",
        "server": "7452",
        "farm": 8,
        "title": "Welcome to Middle Earth - Explored",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "23958447389",
        "owner": "76482173@N05",
        "secret": "c6ed281ea5",
        "server": "1567",
        "farm": 2,
        "title": "Watagan Lightning",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "26917893065",
        "owner": "32553078@N08",
        "secret": "0696225a3e",
        "server": "7620",
        "farm": 8,
        "title": "Sunset on Place de l'Etoile after a rainy day",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "29558462534",
        "owner": "86832534@N03",
        "secret": "7ac1239618",
        "server": "8554",
        "farm": 9,
        "title": "Ethereal (Explored)",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "25017412802",
        "owner": "26248101@N08",
        "secret": "d26a2b206a",
        "server": "1559",
        "farm": 2,
        "title": "Iron Maiden",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    },
    {
        "id": "25788014884",
        "owner": "130608600@N05",
        "secret": "4d19109166",
        "server": "1633",
        "farm": 2,
        "title": "CRS-8 first stage landing",
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0,
        "is_primary": 0,
        "has_comment": 0
    }
];

export const gallery = {
    "id": "66911286-72157677476597845",
    "gallery_id": "72157677476597845",
    "url": "https://www.flickr.com/photos/flickr/galleries/72157677476597845",
    "owner": "66956608@N06",
    "username": "Flickr",
    "iconserver": "3741",
    "iconfarm": 4,
    "primary_photo_id": "24173905193",
    "date_create": "1480995213",
    "date_update": "1514652568",
    "count_photos": 48,
    "count_videos": 0,
    "count_total": 48,
    "count_views": 38412,
    "count_comments": 69,
    "title": {
        "_content": "Top Flickr Photos of 2016"
    },
    "description": {
        "_content": "For the Top Flickr Photos of 2016, we sorted through billions of uploads to arrive at these 50 favorites. \n\nTo compile this list, we started with an algorithm that calculates a combination of social and engagement metrics. To avoid the results being merely a popularity contest, the selection also involved curation by Flickr staff.\n\nWhere photos from the same community members appeared multiple times in the algorithmic ranking, we‚Äôve selected one of their top-scoring images to represent their work.\n\n<a href=\"http://blog.flickr.net/en/2016/12/06/top-flickr-photos-from-2016/\" rel=\"nofollow\">blog.flickr.net/en/2016/12/06/top-flickr-photos-from-2016/</a>"
    },
    "sort_group": null,
    "primary_photo_server": "1695",
    "primary_photo_farm": 2,
    "primary_photo_secret": "89196bf532",
    "owner_name": "Flickr "
}

export const image = {
    "id": "24679948091",
    "secret": "dd3e3f117f",
    "server": "1671",
    "farm": 2,
    "dateuploaded": "1454423486",
    "isfavorite": 0,
    "license": "0",
    "safety_level": "0",
    "rotation": 0,
    "owner": {
        "nsid": "62118446@N03",
        "username": "cbrutel",
        "realname": "Christophe Brutel",
        "location": "Espoo, Finland",
        "iconserver": "7226",
        "iconfarm": 8,
        "path_alias": "christophebrutel"
    },
    "title": {
        "_content": "Braving the cold"
    },
    "description": {
        "_content": "Haukilahti, Espoo, Finland\n\nShot with Panasonic DMC-LX100, processed with Lightroom CC 2015."
    },
    "visibility": {
        "ispublic": 1,
        "isfriend": 0,
        "isfamily": 0
    },
    "dates": {
        "posted": "1454423486",
        "taken": "2016-01-05 09:13:59",
        "takengranularity": "0",
        "takenunknown": "0",
        "lastupdate": "1537079823"
    },
    "views": "4429063",
    "editability": {
        "cancomment": 0,
        "canaddmeta": 0
    },
    "publiceditability": {
        "cancomment": 1,
        "canaddmeta": 0
    },
    "usage": {
        "candownload": 0,
        "canblog": 0,
        "canprint": 0,
        "canshare": 0
    },
    "comments": {
        "_content": "612"
    },
    "notes": {
        "note": [
            {
                "id": "72157664079284881",
                "photo_id": "24679948091",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "authorrealname": "Christophe Brutel",
                "authorispro": 1,
                "authorisdeleted": 0,
                "x": "212",
                "y": "8",
                "w": "27",
                "h": "27",
                "_content": "winter"
            },
            {
                "id": "72157663534634689",
                "photo_id": "24679948091",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "authorrealname": "Christophe Brutel",
                "authorispro": 1,
                "authorisdeleted": 0,
                "x": "262",
                "y": "8",
                "w": "27",
                "h": "27",
                "_content": "swan"
            },
            {
                "id": "72157664047992672",
                "photo_id": "24679948091",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "authorrealname": "Christophe Brutel",
                "authorispro": 1,
                "authorisdeleted": 0,
                "x": "262",
                "y": "40",
                "w": "27",
                "h": "27",
                "_content": "Finland"
            },
            {
                "id": "72157661906899043",
                "photo_id": "24679948091",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "authorrealname": "Christophe Brutel",
                "authorispro": 1,
                "authorisdeleted": 0,
                "x": "212",
                "y": "40",
                "w": "27",
                "h": "27",
                "_content": "Espoo"
            },
            {
                "id": "72157664176344186",
                "photo_id": "24679948091",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "authorrealname": "Christophe Brutel",
                "authorispro": 1,
                "authorisdeleted": 0,
                "x": "212",
                "y": "70",
                "w": "27",
                "h": "27",
                "_content": "ice"
            },
            {
                "id": "72157664079307221",
                "photo_id": "24679948091",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "authorrealname": "Christophe Brutel",
                "authorispro": 1,
                "authorisdeleted": 0,
                "x": "262",
                "y": "70",
                "w": "27",
                "h": "27",
                "_content": "Baltic"
            }
        ]
    },
    "people": {
        "haspeople": 0
    },
    "tags": {
        "tag": [
            {
                "id": "62095392-24679948091-267406",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Aube",
                "_content": "aube",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-9837",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Baltic",
                "_content": "baltic",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-433049",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Baltique",
                "_content": "baltique",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-594",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Bird",
                "_content": "bird",
                "machine_tag": 0
            },
            {
                "id": "62095392-24679948091-87231",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Brume",
                "_content": "brume",
                "machine_tag": 0
            },
            {
                "id": "62095392-24679948091-64137",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Cygne",
                "_content": "cygne",
                "machine_tag": 0
            },
            {
                "id": "62095392-24679948091-45578685",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Cygne tubercule",
                "_content": "cygnetubercule",
                "machine_tag": 0
            },
            {
                "id": "62095392-24679948091-1998",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Dawn",
                "_content": "dawn",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-1430",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Espoo",
                "_content": "espoo",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-1436",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Finland",
                "_content": "finland",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-3342",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Frost",
                "_content": "frost",
                "machine_tag": 0
            },
            {
                "id": "62095392-24679948091-2284",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Frozen",
                "_content": "frozen",
                "machine_tag": 0
            },
            {
                "id": "62095392-24679948091-50693",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Gel",
                "_content": "gel",
                "machine_tag": 0
            },
            {
                "id": "62095392-24679948091-96972",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Glace",
                "_content": "glace",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-2302284",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Haukilahti",
                "_content": "haukilahti",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-6074",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Hiver",
                "_content": "hiver",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-682",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Ice",
                "_content": "ice",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-96",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Leica",
                "_content": "leica",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-134742",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Lever de Soleil",
                "_content": "leverdesoleil",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-381",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Mer",
                "_content": "mer",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-398",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Mist",
                "_content": "mist",
                "machine_tag": 0
            },
            {
                "id": "62095392-24679948091-155961",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Mute swan",
                "_content": "muteswan",
                "machine_tag": 0
            },
            {
                "id": "62095392-24679948091-52410",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Oiseau",
                "_content": "oiseau",
                "machine_tag": 0
            },
            {
                "id": "62095392-24679948091-11243",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Panasonic",
                "_content": "panasonic",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-228",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Sea",
                "_content": "sea",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-1370",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Sunrise",
                "_content": "sunrise",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-2752",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Swan",
                "_content": "swan",
                "machine_tag": 0
            },
            {
                "id": "62095392-24679948091-201",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "Winter",
                "_content": "winter",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-9993892",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "lx100",
                "_content": "lx100",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-392175",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "2016",
                "_content": "2016",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-7328",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "january",
                "_content": "january",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-121",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "travel",
                "_content": "travel",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-791",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "nature",
                "_content": "nature",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-3272",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "europe",
                "_content": "europe",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-544",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "color",
                "_content": "color",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-545",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "colour",
                "_content": "colour",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-2326",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "photo",
                "_content": "photo",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-4634",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "photos",
                "_content": "photos",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-1935",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "photography",
                "_content": "photography",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-3444",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "raw",
                "_content": "raw",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-1695",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "landscape",
                "_content": "landscape",
                "machine_tag": 0
            },
            {
                "id": "62095392-24679948091-441",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "seascape",
                "_content": "seascape",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-168898",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "waterscape",
                "_content": "waterscape",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-800",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "water",
                "_content": "water",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-4890",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "outdoor",
                "_content": "outdoor",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-2077205",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "explored",
                "_content": "explored",
                "machine_tag": false
            },
            {
                "id": "62095392-24679948091-11820",
                "author": "62118446@N03",
                "authorname": "cbrutel",
                "raw": "explore",
                "_content": "explore",
                "machine_tag": false
            }
        ]
    },
    "location": {
        "latitude": "60.154233",
        "longitude": "24.776336",
        "accuracy": "16",
        "context": "0",
        "locality": {
            "_content": "Espoo",
            "place_id": "0dYvnjJUVL7l.nI",
            "woeid": "564617"
        },
        "county": {
            "_content": "Helsingin",
            "place_id": "paE.WWFUVLvDkOxqOA",
            "woeid": "56120851"
        },
        "region": {
            "_content": "Uusimaa",
            "place_id": "_7C9msNTW7n35YTFQQ",
            "woeid": "29389247"
        },
        "country": {
            "_content": "Finland",
            "place_id": "5ryEXkVTUb66Tnx33g",
            "woeid": "23424812"
        },
        "place_id": "0dYvnjJUVL7l.nI",
        "woeid": "564617"
    },
    "geoperms": {
        "ispublic": 1,
        "iscontact": 0,
        "isfriend": 0,
        "isfamily": 0
    },
    "urls": {
        "url": [
            {
                "type": "photopage",
                "_content": "https://www.flickr.com/photos/christophebrutel/24679948091/"
            }
        ]
    },
    "media": "photo"
};