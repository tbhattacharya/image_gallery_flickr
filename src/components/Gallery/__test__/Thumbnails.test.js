import React from 'react';
import { photo } from '../../../utils/test-utils/GalleryData';
import { withStyles } from '@material-ui/core/styles';
import { createRender } from '@material-ui/core/test-utils';
import Thumbnails from '../Thumbnails';

describe("Testing GalleryHeader", function () {
    const render = createRender();
    const mockCallback = jest.fn();
    let wrapper;
    beforeEach(() => {
        wrapper = render(<Thumbnails key={photo[0].id} image={photo[0]} handleClick = {()=>{mockCallback()}} />)
    })

    it("Expect Thumbnail to be drawn without exception", function () {
        expect(wrapper).toBeDefined();
    });


    it("Expect Thumbnail to match snapshot", function () {
        expect(wrapper).toMatchSnapshot();
    });
});