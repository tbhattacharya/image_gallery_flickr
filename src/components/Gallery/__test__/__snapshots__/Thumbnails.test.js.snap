// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GalleryHeader Expect Thumbnail to match snapshot 1`] = `
<div
  class="MuiGrid-item-08 MuiGrid-grid-xs-3-038 Thumbnails-thumbnail-05"
>
  <div
    class="Thumbnails-overlayItem-04"
  >
    Cuando amanece la pasi√≥n
  </div>
  <div
    class="Thumbnails-photo-06"
  >
    <span
      class="undefined lazy-load-image-background blur"
      style="background-image:url( undefined);background-size:100% 100%;color:transparent;display:inline-block;height:100%;width:100%"
    >
      <span
        class=""
        style="display:inline-block;height:100%;width:100%"
      >
        
      </span>
    </span>
  </div>
</div>
`;
